extensions/data-plane-selector/data-plane-selector-api/src/main/java/org/eclipse/edc/connector/dataplane/selector/transformer/JsonObjectToDataPlaneInstanceTransformer.java

```diff
@@ -0,0 +1,96 @@
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.selector.transformer;
+
+import jakarta.json.JsonArray;
+import jakarta.json.JsonNumber;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonValue;
+import org.eclipse.edc.connector.dataplane.selector.spi.instance.DataPlaneInstance;
+import org.eclipse.edc.jsonld.spi.transformer.AbstractJsonLdTransformer;
+import org.eclipse.edc.transform.spi.TransformerContext;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import static org.eclipse.edc.connector.dataplane.selector.spi.instance.DataPlaneInstance.ALLOWED_DEST_TYPES;
+import static org.eclipse.edc.connector.dataplane.selector.spi.instance.DataPlaneInstance.ALLOWED_SOURCE_TYPES;
+import static org.eclipse.edc.connector.dataplane.selector.spi.instance.DataPlaneInstance.Builder;
+import static org.eclipse.edc.connector.dataplane.selector.spi.instance.DataPlaneInstance.LAST_ACTIVE;
+import static org.eclipse.edc.connector.dataplane.selector.spi.instance.DataPlaneInstance.PROPERTIES;
+import static org.eclipse.edc.connector.dataplane.selector.spi.instance.DataPlaneInstance.TURNCOUNT;
+import static org.eclipse.edc.connector.dataplane.selector.spi.instance.DataPlaneInstance.URL;
+
+public class JsonObjectToDataPlaneInstanceTransformer extends AbstractJsonLdTransformer<JsonObject, DataPlaneInstance> {
+    public JsonObjectToDataPlaneInstanceTransformer() {
+        super(JsonObject.class, DataPlaneInstance.class);
+    }
+
+    @Override
+    public @Nullable DataPlaneInstance transform(@NotNull JsonObject jsonObject, @NotNull TransformerContext context) {
+        var builder = Builder.newInstance();
+        builder.id(nodeId(jsonObject));
+        visitProperties(jsonObject, (key, jsonValue) -> transformProperties(key, jsonValue, builder, context));
+
+        return builder.build();
+    }
+
+    private void transformProperties(String key, JsonValue jsonValue, DataPlaneInstance.Builder builder, TransformerContext context) {
+        switch (key) {
+            case URL -> {
+                try {
+                    builder.url(new URL(Objects.requireNonNull(transformString(jsonValue, context))));
+                } catch (MalformedURLException e) {
+                    context.reportProblem(e.getMessage());
+                }
+            }
+            case LAST_ACTIVE -> transformLong(context, jsonValue, builder::lastActive);
+            case TURNCOUNT -> builder.turnCount(transformInt(jsonValue, context));
+            case ALLOWED_DEST_TYPES -> {
+                var obj = transformArray(jsonValue, Object.class, context);
+                if (obj != null && !obj.isEmpty()) {
+                    builder.allowedDestTypes(Set.copyOf(obj.stream().map(Object::toString).toList()));
+                }
```

Comment:
ad `toString`: the problem is, we don't have a `JsonValueToString` transformer: after unwrapping the `JsonArray`, it delegates back to the context and tries to find a transformer capable of `JsonObject -> String`, which we don't have. 

ad source/dest: we don't enforce it on the entity, so i think it's valid as it is. For example, there could be a custom strategy, that interprets a null value as "can handle all"

Reply:
generally when there are lists/sets/maps involved I'd prefer to avoid nulls, the empty data structure generally does the job, anyway, let's keep it like this for now.